### ISSUE TITLE
[raspberrypi4][kirkstone] Can't reconnect to my CM4 Bluetooth after removing it from whitelist

### ISSUE THREAD

**Issue by Piocky** [agherzan/meta-raspberrypi#1135]
Hello,

I recently moved from dunfell to kirkstone, and I've got a weird behavior that I cannot explain:
- on dunfell, I was able to connect to my CM4 via a phone to play audio via Bluetooth, removing it from the known devices of the phone, and reconnect via the pairing process again later.
- on kirkstone, if I do the pairing process once, then remove the CM4 from the known devices, next time I try to redo the pairing process its fails and my phone prompt a message saying "connection impossible, forget this device", but then I'm stuck. The only way I can connect again is by removing the info file (/var/lib/bluetooth/{address of my cm4}/{address of my phone}/info, and then restart bluetooth services.

Does anyone know what change could have affected this behavior? I'm on the same bluez version, as well as bluez-firmware-rpidistro.
Attached are the logs of the failure connection.

[BT connection failed.txt](https://github.com/agherzan/meta-raspberrypi/files/10360876/BT.connection.failed.txt)

**Comment by agherzan** [comment]
I haven't seen this. Are you able to reproduce this on other boards? Is RaspberryPi OS also affected?

**Comment by Piocky** [comment]
It happens on all my CM4 with my new image, I tried to rollback to previous bluez and bluez-firmware versions, but it doesn't change anything.. My guess is that it's somehow related to the Link Key, which should be generated again when a new connection occurs, and then rewrited on dunfell, but not on Kirkstone.
I didn't try on Raspbian, so can tell about that.

**Comment by Piocky** [comment]
@agherzan I just tested it on my dev Raspberry Pi which is on Raspbian, and I have the exact same behavior

**Comment by agherzan** [comment]
In that case, I would report that to rpi directly. Sounds like a firmware issue.

**Comment by Piocky** [comment]
I've got some more inputs to provide:
I just did some tests using the [simple-agent](https://git.kernel.org/pub/scm/bluetooth/bluez.git/tree/test/simple-agent) script file provide by bluez, and I get the error only when I choose NoInputNoOutput for the capability of the agent. All the other methods work well. The annoying thing is that my device doesn't have any input or output capability, so I can't use anything else than NoInputNoOutput, but at least I narrowed down the issue.
